File: 941-sort-array-by-parity/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int i = 0, j=0;
        while(i<nums.size())
            {
                if(nums[i]%2==0)
                    {
                        swap(nums[i],nums[j]);
                        i++;
                        j++;
                    }
                else
                    {
                        i++;

                    }
            }
            return nums;
    }
};

File: 941-sort-array-by-parity/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        k=0
        i=0
        for i in range(len(nums)):
            if nums[i]%2==0:
                nums[k],nums[i]=nums[i],nums[k]
                k+=1
            else:
                continue

        return nums
        